src/transformers/models/patchtst/configuration_patchtst.py

```diff
@@ -0,0 +1,270 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""PatchTST model configuration"""
+
+from typing import List, Optional, Union
+
+from transformers.configuration_utils import PretrainedConfig
+from transformers.utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+PATCHTST_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "ibm/patchtst-base": "https://huggingface.co/ibm/patchtst-base/resolve/main/config.json",
+    # See all PatchTST models at https://huggingface.co/ibm/models?filter=patchtst
+}
+
+
+class PatchTSTConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of an [`PatchTSTModel`]. It is used to instantiate an
+    PatchTST model according to the specified arguments, defining the model architecture.
+    [ibm/patchtst](https://huggingface.co/ibm/patchtst) architecture.
+
+    Configuration objects inherit from [`PretrainedConfig`] can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+    Args:
+        num_input_channels (`int`, *optional*, defaults to 1):
+            The size of the target variable which by default is 1 for univariate targets. Would be > 1 in case of
+            multivariate targets.
+        context_length (`int`, *optional*, defaults to 32):
+            The context length for the encoder.
+        distribution_output (`str`, *optional*, defaults to `"student_t"`):
+            The distribution emission head for the model when loss is "nll". Could be either "student_t", "normal" or
+            "negative_binomial".
+        loss (`str`, *optional*, defaults to `"mse"`):
+            The loss function for the model corresponding to the `distribution_output` head. For parametric
+            distributions it is the negative log likelihood ("nll") and for point estimates it is the mean squared
+            error "mse".
+        patch_length (`int`, *optional*, defaults to 1):
+            Define the patch length of the patchification process. Default to 1.
+        stride (`int`, *optional*, defaults to 1):
```

Comment:
Or something similar - just so it's clearer when modifying the config e.g. `config.patch_stride = 1` what is being affected. As it's not really the stride of the patch, happy for an alternative or to leave as is 

```suggestion
        patch_stride (`int`, *optional*, defaults to 1):
```